File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Northwind.RegionForm.ts
namespace Serene.Northwind {
    export class RegionForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Region';

    }

    export interface RegionForm {
        RegionID: Serenity.IntegerEditor;
        RegionDescription: Serenity.StringEditor;
    }

    [['RegionID', () => Serenity.IntegerEditor], ['RegionDescription', () => Serenity.StringEditor]].forEach(x => Object.defineProperty(RegionForm.prototype, <string>x[0], { get: function () { return this.w(x[0], (x[1] as any)()); }, enumerable: true, configurable: true }));
}

File: Serene/Serene.Script/Imports/ServerImports/Northwind.RegionForm.cs
using jQueryApi;
using Serenity;
using Serenity.ComponentModel;
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Runtime.CompilerServices;

namespace Serene.Northwind
{
    public partial class RegionForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Region";

        public RegionForm(string idPrefix) : base(idPrefix) {}

        public IntegerEditor RegionID { [InlineCode("{this}.w('RegionID', Serenity.IntegerEditor)")] get; private set; }
        public StringEditor RegionDescription { [InlineCode("{this}.w('RegionDescription', Serenity.StringEditor)")] get; private set; }
    }
}

